lib/widgets/home.dart

```diff
@@ -307,29 +308,104 @@ void _showMainMenu(BuildContext context, {
+  });
+
+  final List<Widget> menuItems;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+
+    return SafeArea(
+      minimum: const EdgeInsets.only(bottom: 8),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          _MainMenuHeader(),
+          Flexible(child: InsetShadowBox(
+            top: 8, bottom: 8,
+            color: designVariables.bgBotBar,
+            child: SingleChildScrollView(
+              padding: const EdgeInsets.symmetric(vertical: 8, horizontal: 8),
+              child: Column(children: menuItems)))),
+          const Padding(
+            padding: EdgeInsets.symmetric(horizontal: 16),
+            child: AnimatedScaleOnTap(
+              scaleEnd: 0.95,
+              duration: Duration(milliseconds: 100),
+              child: BottomSheetDismissButton(
+                style: BottomSheetDismissButtonStyle.close))),
+        ]));
+  }
+}
+
+class _MainMenuHeader extends StatelessWidget {
+  const _MainMenuHeader();
+
+  @override
+  Widget build(BuildContext context) {
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+
+    final realmIconUrl = store.tryResolveUrl(store.realmIcon.toString());
+
+    return Padding(
+      padding: const EdgeInsets.only(top: 6),
+      child: Row(children: [
+        Expanded(child: Padding(
+          padding: const EdgeInsets.fromLTRB(12, 6, 4, 6),
+          child: Row(spacing: 8, children: [
+            AvatarShape(
+              size: 28,
+              borderRadius: 4,
+              child: realmIconUrl != null
+                ? RealmContentNetworkImage(realmIconUrl)
+                : const SizedBox.shrink()),
+            Expanded(child: Text(store.realmName,
+              overflow: TextOverflow.ellipsis,
+              style: TextStyle(
+                color: designVariables.title,
+                fontSize: 20,
+                height: 24 / 20,
+              ).merge(weightVariableTextStyle(context, wght: 600)))),
+          ]))),
+        AnimatedScaleOnTap(
+          duration: const Duration(milliseconds: 100),
+          scaleEnd: 0.95,
+          child: TextButton(
+            onPressed: () {
+              Navigator.pop(context); // Close the main menu.
+              Navigator.of(context).push(
+                MaterialWidgetRoute(page: const ChooseAccountPage()));
+            },
+            style: TextButton.styleFrom(
+              padding: const EdgeInsets.fromLTRB(8, 7, 14, 7),
+              foregroundColor: designVariables.icon,
+              backgroundColor: Colors.transparent,
+              overlayColor: Colors.transparent,
+              splashFactory: NoSplash.splashFactory,
+              shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
```

Comment:
This border radius shouldn't make a difference, with a transparent background/overlay/etc., right? I see it in the Figma, but it looks accidental there, too; I think we can leave it out without comment.